### DISCUSSION TITLE
HASPI Usage Standards For Evaluation Reporting

### DISCUSSION THREAD

**Discussion by jhkonan** [claritychallenge/clarity#140] [Q&A] [answered]
Hello, I am a researcher on speech enhancement and I am new to the use of objective intelligibility measures in the area of hearing impairment. I was able to figure out how to use MBSTOI, but HASPI has multiple parameters that I am not sure how to specify.

What are the standard specifications used for HASPI? Specifically clarity.evaluator.haspi.haspi_v2.
https://github.com/claritychallenge/clarity/blob/main/clarity/evaluator/haspi/haspi.py

Here is my attempt making an example:

```
! pip3 install https://github.com/claritychallenge/clarity/archive/refs/heads/main.zip --quiet
from clarity.evaluator import haspi, mbstoi

import os

root_url  = "https://github.com/microsoft/DNS-Challenge/raw/interspeech2020/master/datasets/test_set/synthetic/no_reverb"
clean_url = root_url + "/clean/clean_fileid_0.wav"
noisy_url = root_url + "/noisy/clnsp126_3Wjw0nadnM4_snr15_tl-22_fileid_0.wav"

if "clean.wav" not in os.listdir() or "noisy.wav" not in os.listdir():
    os.system("wget -O clean.wav " + clean_url)
    os.system("wget -O noisy.wav " + noisy_url)

import librosa

clean = librosa.load("clean.wav", sr=16000)[0]
noisy = librosa.load("noisy.wav", sr=16000)[0]

Y1 = mbstoi.mbstoi(clean, clean, noisy, noisy, fs_signal=16000)
Y2, _ = haspi.haspi_v2(clean, 16000, noisy, 16000, [250, 500, 1000, 2000, 4000, 6000])

print("MBSTOI ", Y1, "\n", "HASPI  ", Y2, sep="")
```

However, the observed HASPI value is really small. Is this correct usage? What are the standard ways to specify the evaluation?

**Comment by TuZehai** [comment, marked as answer]
Hi jhkonan,

The parameter _HL_ in _haspi_v2_ should be the hearing losses at [250, 500, 1000, 2000, 4000, 6000] Hz, e.g., [10, 20, 30, 40, 50, 60] dB, or [0] * 6 for normal hearing.

Zehai
